File: September-9-2025/maxConsecutiveOnes.js
// Given a binary array nums and an integer k, return the maximum number of consecutive 1's in the array if you can flip at most k 0's.

// Input: nums = [1,1,1,0,0,0,1,1,1,1,0], k = 2
// Output: 6
// Explanation: [1,1,1,0,0,1,1,1,1,1,1]
// Bolded numbers were flipped from 0 to 1. The longest subarray is underlined.

var nums = [1,1,1,0,0,0,1,1,1,1,0], k = 2


const maxConsecutiveOnes = (nums, k) => { 
  var zeroesCount = 0; 
  var max = 0; 
  var left = 0; 

  for(let right = 0; right < nums.length; right++){
   
    if(nums[right] === 0){
      zeroesCount++
    }

    while(zeroesCount > k){
      if(nums[left] === 0){
        zeroesCount--
      }
      left++
    }
    max = Math.max(max, right - left + 1)
  }
  return max
}

console.log(maxConsecutiveOnes(nums, k))

File: 2026/January-9-2026/maxConsecutiveOnes.py
# Given a binary array nums and an integer k, return the maximum number of consecutive 1's in the array if you can flip at most k 0's.

# Example 1:

# Input: nums = [1,1,1,0,0,0,1,1,1,1,0], k = 2
# Output: 6
# Explanation: [1,1,1,0,0,1,1,1,1,1,1]
# Bolded numbers were flipped from 0 to 1. The longest subarray is underlined.
# Example 2:

from typing import List

nums = [1,1,1,0,0,0,1,1,1,1,0]
k = 2

def max_consecutive_ones(nums: List[int], k: int) -> int: 
  left = 0
  zeroes_count = 0
  max_len = 0

  for right in range(len(nums)): 
    if nums[right] == 0: 
      zeroes_count+= 1

    while zeroes_count > k: 
      if nums[left] == 0: 
        zeroes_count-=1
      left += 1
    max_len = max(max_len, right - left + 1)
  return max_len 


print(max_consecutive_ones(nums, k))
    
